synapse/handlers/ui_auth/checkers.py

```diff
@@ -49,7 +49,7 @@ async def check_auth(self, authdict: dict, clientip: str) -> Any:
             clientip: The IP address of the client.
 
         Raises:
-            SynapseError if authentication failed
+            LoginError if authentication failed.
 
         Returns:
             The result of authentication (to pass back to the client?)
```

Comment:
I don't suppose we can clarify this comment while we're looking at all of this!

Reply:
Clarify in what way? Say why/how LoginError is used?

Reply:
> (to pass back to the client?)

It seems whoever wrote this comment wasn't sure why we care about this. If you've groked this and know the answer it would be great to say *why* we care about the result of the authentication.

Reply:
The only checkers that return something other than just `True` are the threepid ones:
https://github.com/matrix-org/synapse/blob/8c654b73095a594b36101aa81cf91a8e1bebc29f/synapse/handlers/ui_auth/checkers.py#L206

And the only place I can see the result is accessed is here, ~~but I have no clue where the user ID would get into the result field of a UIA stage.~~ It looks like the [password provider docs](https://matrix-org.github.io/synapse/latest/password_auth_providers.html#optional-methods) (which you wrote?) mention that 
https://github.com/matrix-org/synapse/blob/8c654b73095a594b36101aa81cf91a8e1bebc29f/synapse/handlers/auth.py#L377-L405

Where `result` comes from the only ever use of `get_completed_ui_auth_stages`:
https://github.com/matrix-org/synapse/blob/8c654b73095a594b36101aa81cf91a8e1bebc29f/synapse/handlers/auth.py#L605

There's also a section in the SSO handler which sets the result field, but I'm not sure if that's ever looked at later:
https://github.com/matrix-org/synapse/blob/8c654b73095a594b36101aa81cf91a8e1bebc29f/synapse/handlers/sso.py#L690-L704

I am currently using the fact that the result is stored for expiring registration tokens, but I think I could use the UIA session data instead.
https://github.com/govynnus/synapse/blob/e9435f8f6c9ba1497f689e3963c9c005c1d71dfc/synapse/handlers/ui_auth/checkers.py#L294
https://github.com/govynnus/synapse/blob/e9435f8f6c9ba1497f689e3963c9c005c1d71dfc/synapse/storage/databases/main/ui_auth.py#L333-L373

So basically no, I don't grok it.
Although I'm pretty sure it's not passed back to the client :-)

Reply:
Hah, thanks for taking another look! We can just leave it as is then!